controllers/ItemsController.php

```diff
@@ -988,6 +992,11 @@ private function _handleFileRequest($item) {
 				
 				$info = Zotero_Storage::getUploadInfo($uploadKey);
 				if (!$info) {
+					if(Z_Core::$MC->get("successfulUploadKey_" . $uploadKey)) {
+						header("HTTP/1.1 204 No Content");
+						header("Last-Modified-Version: " . $item->version);
+						exit;
+					}
```

Comment:
The problem with this is that memcached should really be optional, and if reads or writes were failing for some reason (which happens sometimes) this would mean that all uploads from 4.0 would start producing errors.

Maybe for the time being we should add a `processed` boolean column to `storageUploadQueue` (with an index on it), set it to `1` on registration, and clear out processed entries after a few minutes with the event scheduler?